src/libstd/io/mod.rs

```diff
@@ -322,6 +576,17 @@ pub trait Write {
     ///
     /// It is **not** considered an error if the entire buffer could not be
     /// written to this writer.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::io::Write;
+    ///
+    /// let mut buffer = Vec::new();
```

Comment:
it was intentional in the sense that i mis-read the docs and thought that it was a little strange that vecs couldn't implement `Read`, but kinda just pushed ahead rather than digging deeper. you'd prefer the all use files? I was going to say that it'd be better to use all vecs, as at least the tests would run...